ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/ImageSingleVulnerabilities.tsx

```diff
@@ -100,34 +83,35 @@ function ImageSingleVulnerabilities({ imageId }: ImageSingleVulnerabilitiesProps
                 </EmptyState>
             </Bullseye>
         );
-    } else if (loading && !data) {
+    } else if (loading && !vulnerabilityData) {
         mainContent = (
             <Bullseye>
                 <Spinner isSVG />
             </Bullseye>
         );
-    } else if (data) {
-        const vulnerabilities = data.image.imageVulnerabilities;
-        const severityCounts = severityCountsFromImageVulnerabilities(vulnerabilities);
-        const cveStatusCounts = statusCountsFromImageVulnerabilities(vulnerabilities);
+    } else if (vulnerabilityData) {
+        const vulnerabilities = vulnerabilityData.image.imageVulnerabilities;
+
         // TODO Integrate these with page search filters
         const hiddenSeverities = new Set<VulnerabilitySeverity>([]);
         const hiddenStatuses = new Set<FixableStatus>([]);
```

Comment:
Can you share this at next team meeting?

Map and Set remove a security concern about overwriting Object properties.

Reply:
Sure and I can share the `useSet` hook since it might be of use in other places in the app.

What's the security concern you are talking about in this context?

Reply:
Whether possible (see below) or implausible, huge number of reported vulnerabilities are about possible object prototype pollution. Map and Set remove it as even a theoretical risk.

https://github.com/ljharb/qs/pull/428

```js
} else if (cleanRoot !== '__proto__') {
    obj[cleanRoot] = leaf;
}
```